sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,776 @@
+    {
+        runtime: (linear n u1 u679),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_var (n uint))
+    {
+        runtime: (linear n u1 u723),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_contract_storage (n uint))
+    {
+        runtime: (linear n u13 u8043),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_block_info (n uint))
+    {
+        runtime: u5886,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_stx_balance (n uint))
+    {
+        runtime: u1386,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_stx_transfer (n uint))
+    {
+        runtime: u1444,
+        write_length: u1,
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_mint (n uint))
+    {
+        runtime: u1624,
+        write_length: u1,
+        write_count: u2,
+        read_count: u2,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_transfer (n uint))
+    {
+        runtime: u563,
+        write_length: u1,
+        write_count: u2,
+        read_count: u2,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_balance (n uint))
+    {
+        runtime: u543,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_nft_mint (n uint))
+    {
+        runtime: (linear n u1 u724),
+        write_length: u1,
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_nft_transfer (n uint))
+    {
+        runtime: (linear n u1 u787),
+        write_length: u1,
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_nft_owner (n uint))
+    {
+        runtime: (linear n u1 u680),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_get_supply (n uint))
+    {
+        runtime: u474,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_burn (n uint))
+    {
+        runtime: u599,
+        write_length: u1,
+        write_count: u2,
+        read_count: u2,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_nft_burn (n uint))
+    {
+        runtime: (linear n u1 u644),
+        write_length: u1,
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
```

Comment:
Also here, the variable is the NFT Type (basing this on sip-006). Why is runtime different?